homeassistant/components/vizio/media_player.py

```diff
@@ -103,7 +102,10 @@ async def async_setup_entry(
         params["data"] = new_data
 
     if params:
-        hass.config_entries.async_update_entry(config_entry, **params)
+        hass.config_entries.async_update_entry(
+            config_entry,
+            **params,  # type:ignore[arg-type]
```

Comment:
I'm not sure if there is a way to avoid the `type: ignore` comment

Reply:
It's migration code, so I don't think we need to bother too much to find a cleaner way. Preferably we should move the migration to the config flow now since we can set options when creating the config flow result.

Reply:
Then it becomes a much bigger PR - maybe one for the code owners in a follow-up PR?

Reply:
I didn't mean that we should move it here. Just something for the future. This is good as it is.